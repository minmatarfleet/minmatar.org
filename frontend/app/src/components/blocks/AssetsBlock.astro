---
import { i18n } from '@helpers/i18n'
const { t } = i18n(Astro.url)

import type { AssetsLocationUI } from '@dtypes/layout_components'

interface Props {
    assets_location: AssetsLocationUI;
}

const {
    assets_location,
} = Astro.props

import Flexblock from '@components/compositions/Flexblock.astro';

import AssetLocationItem from '@components/blocks/AssetLocationItem.astro';
import AssetItem from '@components/blocks/AssetItem.astro';
import ComponentBlockHeader from '@components/blocks/ComponentBlockHeader.astro';
---

<div
    id="assets-location-list"
    {...Astro.props}
>
    <ComponentBlockHeader>
        <AssetLocationItem
            slot="header"
            asset_location={{
                location_name: assets_location.location_name,
                assets_count: assets_location.assets.reduce( (c, i) => c + i.count , 0 )
            }}
        />

        <Flexblock
            gap='var(--space-3xs)'
            class="[ min-h-[64px] justify-center ]"
            x-show="expanded"
            x-collapse.duration.100ms
        >
            {assets_location.assets.map( (asset) => 
                <AssetItem
                    asset={asset} 
                />
            )}
            {assets_location.assets.length == 0 &&
                t('empty_asset_location')
            }
        </Flexblock>
    </ComponentBlockHeader>
</div>