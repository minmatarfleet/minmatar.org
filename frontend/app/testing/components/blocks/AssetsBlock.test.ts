import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import AssetsBlock from "@components/blocks/AssetsBlock.astro";

test("AssetsBlock defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(AssetsBlock, {
    props: {
      assets_location: {
        location_name: "coop",
        assets: [
          {
            count: 5,
            name: "chickens",
          },
        ],
      },
    },
  });

  expect(result).toMatchSnapshot();
});
