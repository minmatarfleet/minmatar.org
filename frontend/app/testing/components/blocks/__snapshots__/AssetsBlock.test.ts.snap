// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AssetsBlock defaults 1`] = `
"<div id="assets-location-list" assets_location="[object Object]"> <div class="flex-block gap-0" x-data="{
        expanded: true,
        location_assets_count: 0,
        collapse() {
            this.expanded = false
        },
        toggle_collapse() {
            this.expanded = !this.expanded
            global_collapse = false
        },
    }" x-effect="global_collapse &#38;&#38; collapse()" data-astro-cid-xzc67zbc="true" data-mod style="--flex_block_gap: 0;">  <div class="[ sticky-header ]" x-bind:style="\`top: \${post_header_height - 2}px;\`" data-astro-cid-xzc67zbc> <div class="wrapper [ card component ] [ border-bottom ] light-transparency" data-astro-cid-xzc67zbc="true" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-l-xl);--wrapper_padding_inline_end: var(--space-l-xl);--wrapper_padding_block: var(--component-padding-block);--wrapper_max_width: var(--max-text-width);">   <button class="[ !block !w-full ] astro-muoml3se" x-on:click="toggle_collapse()" data-astro-cid-xzc67zbc="true" type="button" data-astro-cid-muoml3se> <div class="fluid-fixed [ items-center ]" data-astro-cid-muoml3se="true" data-mod data-astro-cid-rwsocfon style="--fixed_fluid_gap: var(--space-l);--fixed_fluid_width: 20px;--fixed_fluid_breakpoint: 50%;">   <div class="flex-block [ asset-location-item ] [ w-full ] gap-0" data-mod style="--flex_block_gap: 0;">  <span class="text-[var(--highlight)]"> coop </span> <small> 5 assets in inventory </small>  </div>   <svg width="12" height="8" viewBox="0 0 12 8" fill="none" :class="expanded ? 'expanded' : 'collapsed'" data-astro-cid-muoml3se="true"> <path d="M10.59 0.000156279L6 4.58016L1.41 0.000155477L5.24537e-07 1.41016L6 7.41016L12 1.41016L10.59 0.000156279Z" fill="currentColor"></path> </svg>  </div>   </button>    </div>   </div> <div class="wrapper [ card component ] light-transparency" x-show="expanded" x-collapse.duration.0ms="true" data-astro-cid-xzc67zbc="true" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-l-xl);--wrapper_padding_inline_end: var(--space-l-xl);--wrapper_padding_block: var(--component-padding-block);--wrapper_max_width: var(--max-text-width);">     <div class="flex-block [ min-h-[64px] justify-center ] gap---space-3xs" x-show="expanded" x-collapse.duration.100ms="true" data-mod style="--flex_block_gap: var(--space-3xs);"> <div class="fixed-fluid [ asset-item ]" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 64px;--fixed_fluid_breakpoint: 50%;">  <picture> <img loading="lazy" src="https://images.evetech.net/types/undefined/icon?size=64" width="64" height="64" alt="chickens icon"> </picture> <div class="flex-inline" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-s);--flex_inline_justification: space-between;">  <div class="!flex !items-center">  <div class="flex-block [ w-full ] gap-0" data-mod style="--flex_block_gap: 0;">  <span> 5 x chickens </span>  </div>   </div>  </div>    </div>   </div>     </div>    </div>   </div>"
`;
